msim/src/sim/runtime/mod.rs

```diff
@@ -508,12 +517,40 @@ impl Future for NodeHandle {
     }
 }
 
+/// Checks if current task is killed (or non-existent) - only intended to be used by
+/// return_if_killed!
+pub fn is_current_task_killed() -> bool {
+    if let Some(task) = context::try_current_task() {
+        if task.is_killed() {
+            trace!("current task is killed");
+            return true;
+        }
+    } else {
+        trace!("no current task");
+        return true;
+    }
+    return false;
+}
```

Comment:
nit - take it or leave it: This seems more readable (not having to reason about the fallthrough case) and more idiomatic to me
```suggestion
        if task.is_killed() {
            trace!("current task is killed");
            true
        } else {
            false
        }
    } else {
        trace!("no current task");
        true
    }
}
```

Reply:
fixed